src/sidebar/components/datalist.js

```diff
@@ -0,0 +1,73 @@
+const { createElement } = require('preact');
+const classnames = require('classnames');
+const propTypes = require('prop-types');
+const { useMemo } = require('preact/hooks');
+
+/**
+ * Custom datalist component
+ */
+
+function DataList({
+  activeItem = -1,
+  list,
+  listFormatter = item => item,
+  onSelectItem,
+  open = false,
+}) {
+  const items = useMemo(() => {
+    return list.map((item, index) => {
+      return (
+        <li
+          key={`datalist-${index}`}
+          className={classnames({
+            'is-selected': activeItem === index,
+          })}
+          onClick={() => {
+            onSelectItem(item);
+          }}
```

Comment:
We'll need appropriate ARIA properties to associate the datalist with the input and to indicate the currently selected item in the datalist. I don't know the exact details of the preferred pattern for that. Can you look into it?

Reply:
Sorry, didn't add too many details to this PR last night. I did want to mention I struggled a bit with this as it would end up reading everything in the list and got crazy when I had 50 items. I also observed that the original `<datalist>` was not being read by at least Chrome Vox so I would like to know what the aural  behavior should be here. I'll dig into this more!